src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Number.cs

```diff
@@ -0,0 +1,240 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Buffers.Text
+{
+    public static partial class Utf8Parser
+    {
+        [Flags]
+        private enum ParseNumberOptions
+        {
+            AllowExponent = 0x00000001,
+        }
+
+        private static bool TryParseNumber(ReadOnlySpan<byte> text, ref NumberBuffer number, out int bytesConsumed, ParseNumberOptions options, out bool textUsedExponentNotation)
+        {
+            Debug.Assert(number.Digits[0] == 0 && number.Scale == 0 && !number.IsNegative, "Number not initialized to default(NumberBuffer)");
```

Comment:
Is the Debug.Assert condition and comment inverted here?

Reply:
I know the condition isn't and the MSDN example (and usability) says the text should describe what went wrong.

Reply:
I see. That makes sense.

if number == default, it will be Debug.Assert(true,""), and hence all is good.
if number != default, it will be Debug.Assert(false,""), and we see that message.